os/arch/arm/src/s5j/s5j_boot.c

```diff
@@ -174,28 +174,28 @@ int s5j_mpu_initialize(void)
 	 */
 
 	/* Region 0, Set read only for memory area */
-	mpu_priv_flash(0x0, 0x80000000);
+	mpu_priv_flash(0, 0x0, 0x80000000);
 
 	/* Region 1, for ISRAM(0x02020000++1280KB, RW-WBWA */
-	mpu_user_intsram_wb(S5J_IRAM_PADDR, S5J_IRAM_SIZE);
+	mpu_user_intsram_wb(1, S5J_IRAM_PADDR, S5J_IRAM_SIZE);
 
 	/* Region 2, wifi driver needs non-$(0x02110000++320KB, RW-NCNB */
-	mpu_priv_noncache(S5J_IRAM_PADDR + ((4 + 2 + 8 + 946) * 1024), (320 * 1024));
+	mpu_priv_noncache(2, S5J_IRAM_PADDR + ((4 + 2 + 8 + 946) * 1024), (320 * 1024));
 
 	/* Region 3, for FLASH area, default to set WBWA */
-	mpu_user_intsram_wb(S5J_FLASH_PADDR, S5J_FLASH_SIZE);
+	mpu_user_intsram_wb(3, S5J_FLASH_PADDR, S5J_FLASH_SIZE);
 
 	/* region 4, for Sflash Mirror area to be read only */
-	mpu_priv_flash(S5J_FLASH_MIRROR_PADDR, S5J_FLASH_MIRROR_SIZE);
+	mpu_priv_flash(4, S5J_FLASH_MIRROR_PADDR, S5J_FLASH_MIRROR_SIZE);
 
 	/* Region 5, for SFR area read/write, strongly-ordered */
-	mpu_priv_stronglyordered(S5J_PERIPHERAL_PADDR, S5J_PERIPHERAL_SIZE);
+	mpu_priv_stronglyordered(5, S5J_PERIPHERAL_PADDR, S5J_PERIPHERAL_SIZE);
 
 	/*
 	 * Region 6, for vector table,
 	 * set the entire high vector region as read-only.
 	 */
-	mpu_priv_flash(S5J_IRAM_MIRROR_PADDR, S5J_IRAM_MIRROR_SIZE);
+	mpu_priv_flash(6, S5J_IRAM_MIRROR_PADDR, S5J_IRAM_MIRROR_SIZE);
```

Comment:
1. Why do we need this commit? I don't know what purpose is. It was allocated automatically but you changed it to set manually.
2. You only modified MPU in ARMv7-R. Let's add a location where you make a change in commit title.
```
arch/armv7-r: Modify MPU configuration to accespt region number

Please fill why this is necessary.
```

Reply:
1. When app binary separation is enabled, the mpu configuration api's are called multiple
times during load or restart of an application. The current implementation would assign
new mpu region each time. But we want to change an existing region each time the api is
called. Hence, we modify the api to take the region number and make changes to the region.

2. Done.